game-app/game-core/src/main/java/games/strategy/triplea/ui/BattlePanel.java

```diff
@@ -320,19 +316,33 @@ FightBattleDetails waitForBattleSelection() {
       final Territory unitTerritory,
       final Collection<Territory> territories,
       final boolean noneAvailable) {
-    final Supplier<BombardComponent> action =
-        () -> new BombardComponent(unit, unitTerritory, territories, noneAvailable);
+    final Supplier<SelectTerritoryComponent> action =
+        () -> {
+          final var panel = new SelectTerritoryComponent(unitTerritory, territories, map);
+          final String unitName = unit.getType().getName() + " in " + unitTerritory;
+          panel.setLabelText("Which territory should " + unitName + " bombard?");
+          return panel;
+        };
     return Interruptibles.awaitResult(() -> SwingAction.invokeAndWaitResult(action))
         .result
         .map(
-            comp -> {
-              boolean confirmed = false;
-              while (!confirmed) {
-                confirmed =
-                    EventThreadJOptionPane.showConfirmDialog(
-                        this, comp, "Bombardment Territory Selection", ConfirmDialogType.YES_NO);
+            panel -> {
+              int choice =
+                  EventThreadJOptionPane.showOptionDialog(
+                      this,
+                      panel,
+                      "Bombardment Territory Selection",
+                      JOptionPane.OK_CANCEL_OPTION,
+                      JOptionPane.PLAIN_MESSAGE,
+                      null,
+                      noneAvailable ? new String[] {"OK", "None"} : new String[] {"OK"},
+                      null,
+                      getMap().getUiContext().getCountDownLatchHandler());
+              if (choice != JOptionPane.OK_OPTION) {
+                // User selected the "None" option.
+                return null;
```

Comment:
FIrst I was confused about this because the previous code seems to imply that an action must be selected.
But then I saw your screenshots and noticed that previously "None" was an explicit option, so nice change 👍🏼